### PR TITLE
Fixes #755 Option for deleting a particular item in History

### LINKED ISSUE #755: Option for deleting a particular item in History
**Summary:** 

There is an option  for deleting all the scan history, but there may be some users who might want to delete only a particular product from their history.

**Steps to reproduce:** 

Menu -> History

**Expected behavior:** 

We can improvise the feature by selecting the product we want to delete by long pressing it and then pressing the delete button.

**Screen-shots:** 

A small example of what i'm trying to say.
select-contacts-for-deleting-android

### PR DESCRIPTION
**Pull request by mehroshmmm**
Option for deleting a particular item in History

## Description

Added a swipe controller to handle the swipe events on recycler view

## Related issues and discussion
#755 
 
 ## Checklist
 
 Please make sure these boxes are checked before submitting your pull request - thanks!
 
 
 
 - [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 
 - [ ] If you have multiple commits please combine them into one commit by squashing them.
 
 - [ ] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by policorn** [comment]
Hi @mehroshmmm, I've tested your PR and it provides the functionality described in #755. During testing I've found a small bug (the deletion rectangle is not hidden if you slide back the product to the original position). Here you have a video reproducing it:

pr821

Could you fix it please?

**Comment by Karljoones** [comment]
Can you replace the word delete with a trash can icon? I think it would look better :)

**Comment by mehroshmmm** [comment]
@policorn sure will fix it asap.
@Karljoones thats a good suggestion , will try some trash icons :)

**Comment by policorn** [comment]
HI @mehroshmmm, please use a VectorDrawable to implement the @Karljoones' suggestion

**Comment by mehroshmmm** [comment]
done with the required changes. @policorn should i create a new PR or just commit new changes and you'll fetch them

**Comment by teolemon** [comment]
@mehroshmmm can you fix the small conflict ?
@policorn can you review ?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/HistoryScanActivity.java [hunk 1]
```diff
@@ -210,7 +220,8 @@ protected void onPreExecute() {
 
         @Override
         protected Context doInBackground(Context... ctx) {
-            List<HistoryProduct> listHistoryProducts = mHistoryProductDao.queryBuilder().orderDesc(HistoryProductDao.Properties.LastSeen).list();
+            listHistoryProducts = mHistoryProductDao.queryBuilder().orderDesc(HistoryProductDao.Properties.LastSeen).list();
+            final Bitmap defaultImgUrl = Bitmap.createScaledBitmap(BitmapFactory.decodeResource(getResources(), R.drawable.ic_no), 200, 200, true);
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/HistoryScanActivity.java]: I think Travis CI complains about this resource: `ic_no`

**Comment by teolemon** [comment]
The build is green again @policorn

**Comment by policorn** [comment]
Hi @teolemon; It's ok to merge

**Comment by teolemon** [comment]
Merged, thanks @mehroshmmm :)

**Comment by mehroshmmm** [comment]
@teolemon welcome :)

**Pull request merged**